java/fury-core/src/main/java/org/apache/fury/resolver/ClassResolver.java

```diff
@@ -229,8 +228,7 @@ public class ClassResolver {
   private final ShimDispatcher shimDispatcher;
 
   private static class ExtRegistry {
-    private short registeredClassIdCounter = 0;
-    private final LongMap<Class<?>> registeredId2Classes = new LongMap<>(initialCapacity);
+    private short registeredClassIdCounter = 1;
```

Comment:
why set default value to 1?

Reply:
`registeredClassIdCounter` will increase automatically after each successful registration. When we use the built-in ClassID to register and use `register(Class<?> cls)` again, the `registeredClassIdCounter` can be recalculated once.

```java
// registeredClassIdCounter = 0
register(LambdaSerializer.ReplaceStub.class, LAMBDA_STUB_ID); // registeredClassIdCounter = 1 now
register(class); // registeredClassIdCounter = 1 has been registered at this time, and registeredClassIdCounter needs to be recalculated
```

Reply:
`NO_CLASS_ID` used `0`, is that the reason for use `1` here. Could we add a comment here

Reply:
> `NO_CLASS_ID` used `0`, is that the reason for use `1` here? Could we add a comment here?

Yes, Comment added.